src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage.cs

```diff
@@ -16,7 +16,7 @@ internal abstract class AbstractPackage : AsyncPackage
 
         protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
+            await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(false);
```

Comment:
why was it "true"? was this intentional?

Reply:
> was this intentional?

Yes, package initialization runs on the UI thread. With this change, you introduced two context switches (force switch to thread pool, then immediately switch back to the UI thread). Code that uses JoinableTaskFactory typically does not use `ConfigureAwait`. Our projects require it, so JTF-aware code will show extensive use of `ConfigureAwait(true)` by design.

❗️ This needs to be reverted.

Reply:
Everyone will need to know the rules if we move to JTF. (Technically, everybody should know the rules anyways!)

We will need to be careful where we adopt the rules though: in our case, our ship has sailed that we allowed third parties to call some of our entrypoints without doing a JTF-friendly wait; we will still have to ensure we don't start using JTF in those paths. For something like InitializeAsync where we know exactly who our caller is, it's fine.

Reply:
@heejaechang's question isn't really a JTF question. There is no such thing as a "Joinable Task specific BG". 
The VS package loader may invoke `AsyncPackage.InitializeAsync` from the main thread or a background thread, so you must be prepared for both by switching to the main thread if you need it, and switching off it explicitly if you want to avoid it.

Reply:
@jasonmalinowski @heejaechang Would you folks benefit from a JTF rules meeting where we review the rules briefly and reserve much of the time for the Roslyn team to ask questions and brainstorm on where JTF should be used in Roslyn and where it should be avoided?

As far as the AsyncPackage initialization, using JTF is a must, IMO. Certainly in your rehostable library code you have more freedom to decide. There are indeed places where backward compat can limit your use of JTF. I think it would be good to get a couple folks on the Roslyn team fluent on these cases and then spread that to the rest of the team as needed.